modules/travel_pay/spec/services/claims_service_spec.rb

```diff
@@ -289,26 +287,39 @@
                                                            'end_date' => '2024-03-01T16:45:34Z'
                                                          })
 
-      expect(claims_by_date[:data].count).to equal(0)
-      expect(claims_by_date[:metadata]['status']).to equal(200)
-      expect(claims_by_date[:metadata]['success']).to eq(true)
-      expect(claims_by_date[:metadata]['message']).to eq('No claims found.')
+      expect(claims_by_date.body['data'].count).to equal(0)
+      expect(claims_by_date.body['statusCode']).to equal(200)
+      expect(claims_by_date.body['success']).to eq(true)
+      expect(claims_by_date.body['message']).to eq('No claims found.')
     end
 
-    it 'returns nil if error' do
+    it 'returns original error' do
       allow_any_instance_of(TravelPay::ClaimsClient)
         .to receive(:get_claims_by_date)
         .with(tokens[:veis_token], tokens[:btsss_token], {
                 'start_date' => '2024-01-01T16:45:34Z',
                 'end_date' => '2024-03-01T16:45:34Z'
               })
-        .and_return(claims_error_response)
+        .and_raise(Common::Exceptions::BackendServiceException.new(
+                     'VA900',
+                     { source: 'test' },
+                     401,
+                     {
+                       'statusCode' => 401,
+                       'message' => 'A contact with the specified ICN was not found.',
+                       'success' => false,
+                       'data' => nil
+                     }
+                   ))
 
       claims_by_date = @service.get_claims_by_date_range({
                                                            'start_date' => '2024-01-01T16:45:34Z',
                                                            'end_date' => '2024-03-01T16:45:34Z'
                                                          })
-      expect(claims_by_date).to be_nil
+
+      expect(claims_by_date.body['statusCode']).to equal(401)
+      expect(claims_by_date.body['success']).to eq(false)
+      expect(claims_by_date.body['message']).to eq('A contact with the specified ICN was not found.')
     end
   end
```

Comment:
If we do decide to catch all errors with `rescue e` in the service (see comment above), can we add a test for a non-Faraday::Error like a `NameError` or something?

Reply:
I think I got it? Please double check @kjduensing .

Reply:
To be fair, I just added it after your initial comment, so you didn't actually "miss" it per se. 🤣